virtual_rainforest/models/abiotic_simple/simple_regression.py

```diff
@@ -0,0 +1,677 @@
+    output.append(air_temperature)
+
+    # Mean relative humidity profile, []
+    relative_humidity_log = log_interpolation(
+        data=data,
+        reference_data=data["relative_humidity_ref"].isel(time=time_index),
+        leaf_area_index_sum=leaf_area_index_sum,
+        layer_roles=layer_roles,
+        layer_heights=data["layer_heights"],
+        upper_bound=100,
+        lower_bound=0,
+        gradient=MicroclimateGradients["relative_humidity_gradient"],
+    )
+    relative_humidity = relative_humidity_log.rename("relative_humidity")
+    output.append(relative_humidity)
+
+    # Mean vapor pressure deficit, [kPa]
+    # calculate vapor pressure deficit at reference height first
+    vapor_pressure_deficit_ref = calculate_vapor_pressure_deficit(
+        temperature=data["air_temperature_ref"].isel(time=time_index),
+        relative_humidity=data["relative_humidity_ref"].isel(time=time_index),
+    )
+
+    vapor_pressure_deficit_log = log_interpolation(
+        data=data,
+        reference_data=vapor_pressure_deficit_ref,
+        leaf_area_index_sum=leaf_area_index_sum,
+        layer_roles=layer_roles,
+        layer_heights=data["layer_heights"],
+        upper_bound=10,
+        lower_bound=0,
+        gradient=MicroclimateGradients["vapor_pressure_deficit_gradient"],
+    )
+    vapor_pressure_deficit = vapor_pressure_deficit_log.rename("vapor_pressure_deficit")
+    output.append(vapor_pressure_deficit)
+
+    # Mean soil temperature profile, [C]
+    soil_temperature = interpolate_soil_temperature(
+        layer_heights=data["layer_heights"],
+        layer_roles=layer_roles,
+        surface_temperature=air_temperature.isel(
+            layers=len(layer_roles) - layer_roles.count("soil") - 1
+        ),
+        mean_annual_temperature=data["mean_annual_temperature"],
+    )
+    output.append(soil_temperature)
+
+    # Mean atmospheric pressure profile, [kPa]
+    atmospheric_pressure_1 = xr.concat(
+        [
+            data["atmospheric_pressure_ref"]
+            .isel(time=time_index)
+            .expand_dims(
+                dim={"layers": np.arange(len(layer_roles) - layer_roles.count("soil"))},
+                axis=0,
+            ),  # vertical projection
+            DataArray(
+                np.full((layer_roles.count("soil"), int(data.grid.cell_area)), np.nan),
+                dims=["layers", "cell_id"],
+                coords={
+                    "layers": np.arange(
+                        len(layer_roles) - layer_roles.count("soil"), len(layer_roles)
+                    ),
+                },
+            ),
+        ],
+        dim="layers",
+    )
+    atmospheric_pressure = atmospheric_pressure_1.assign_coords(
+        {
+            "layers": np.arange(0, len(layer_roles)),
+            "layer_roles": ("layers", layer_roles[0 : len(layer_roles)]),
+            "cell_id": data.grid.cell_id,
+        },
+    )
+    output.append(atmospheric_pressure)
```

Comment:
As done in the setup function for the `soil_moisture`, I believe you can do this more easily by taking the `atmospheric_pressure_ref` array (the appropriate time index), coping it, renaming it and then use masking with `where` to set the soil-related layers to `np.nan`.